tests/test_ranges.py

```diff
@@ -1019,6 +1019,15 @@ def test_finite_range(self):
 
 
 class TestFiniteDatetimeRange:
+    def test_cannot_construct_with_inconsistent_tzinfo(self):
```

Comment:
It would be nice to have similar logic for `DatetimeRange` and `HalfFiniteDatetimeRange`, but those two are currently not independent classes but just type aliases. So there is nowhere to insert the logic 🤔 Is there any reason not to make `DatetimeRange` and `HalfFiniteDatetimeRange` actual classes (if we decide to do this I might save that for a separate PR, or do it as a preparation PR)?